airflow/utils/types.py

```diff
@@ -42,7 +43,11 @@ class DagRunType(str, enum.Enum):
     def __str__(self) -> str:
         return self.value
 
-    def generate_run_id(self, logical_date: datetime) -> str:
+    def generate_run_id(self, logical_date: datetime | None, run_after: datetime | None) -> str:
+        if logical_date is None:
+            if run_after is None:
+                raise ValueError("run_after cannot be None")
+            return run_after + get_random_string()
```

Comment:
```suggestion
            return f"{run_after}{get_random_string()}"
```

Reply:
I have changed it here: https://github.com/apache/airflow/pull/46616